util/testutil/addcat/addcat.go

```diff
@@ -0,0 +1,39 @@
+package testutil
+
+import (
+	"bytes"
+	"strings"
+
+	core "github.com/ipfs/go-ipfs/core"
+	coreunix "github.com/ipfs/go-ipfs/core/coreunix"
+	unixfs "github.com/ipfs/go-ipfs/shell/unixfs"
+	errors "github.com/ipfs/go-ipfs/util/debugerror"
+)
+
+func AddCat(adder *core.IpfsNode, catter *core.IpfsNode, data string) error {
```

Comment:
On Sat, May 02, 2015 at 11:20:37AM -0700, Jeromy Johnson wrote:
> because the most common interface for moving data around in go is
> the `io.Reader`, and it reads into a `[]byte` buffer from some
> source (in our data generation case, an RNG)

Sold.  Rerolled using []byte instead of string for AddCat's data.